src/application/redux/reducers/wallet-reducer.ts

```diff
@@ -25,16 +33,27 @@ export function walletReducer(
         masterBlindingKey: payload.masterBlindingKey,
         encryptedMnemonic: payload.encryptedMnemonic,
         passwordHash: payload.passwordHash,
-        confidentialAddresses: payload.confidentialAddresses,
-        utxoMap: {},
+        restorerOpts: payload.restorerOpts,
       };
     }
 
-    case ACTION_TYPES.WALLET_SET_ADDRESS_SUCCESS: {
+    case ACTION_TYPES.NEW_CHANGE_ADDRESS_SUCCESS: {
       return {
         ...state,
-        errors: undefined,
-        confidentialAddresses: state.confidentialAddresses.concat([payload.address]),
+        restorerOpts: {
+          ...state.restorerOpts,
+          lastUsedInternalIndex: state.restorerOpts.lastUsedInternalIndex + 1,
```

Comment:
If you set it at "+ 1" it is not the last one anymore

Reply:
NEW_CHANGE_ADDRESS_SUCCESS is dispatched each time we have generated the `nextChangeAddress`, that's why "+1". What do u mean :thinking: ?